### PR TITLE
Fix the editor button being enabled while the player is dead

### LINKED ISSUE #2045: Make sure editor button is not enabled if the player is dead
still getting new reports that you can go to the editor when dead

### PR DESCRIPTION
**Pull request by GatoCreador887**
Fix #2045.

### CODE REVIEW

File: src/microbe_stage/Microbe.cs [hunk 1]
```diff
@@ -1485,7 +1485,8 @@ private void ReadyToReproduce()
             // The player doesn't split automatically
             allOrganellesDivided = true;
 
-            OnReproductionStatus?.Invoke(this, true);
+            if (!Dead)
+                OnReproductionStatus?.Invoke(this, true);
```
Comment (hhyyrylainen) [inline src/microbe_stage/Microbe.cs:1489]: Wouldn't it be better to have `OnReproductionStatus?.Invoke(this, !Dead);`?

Comment (GatoCreador887) [reply src/microbe_stage/Microbe.cs:1489]: I considered that, but thought it might be a little unclear.

Comment (hhyyrylainen) [reply src/microbe_stage/Microbe.cs:1489]: To me the really unclear thing is why a dead cell is allowed to even call ReadyToReproduce at all.

**Review by hhyyrylainen — APPROVED** [review]
Looks good and seems to work correctly.

**Comment by hhyyrylainen** [comment]
Looks like code style check is not happy (it wants an extra blank line)

**Pull request merged**